sqlglot/dialects/dialect.py

```diff
@@ -498,13 +499,14 @@ def _if_sql(self: Generator, expression: exp.If) -> str:
     return _if_sql
 
 
-def arrow_json_extract_sql(self: Generator, expression: exp.JSONExtract | exp.JSONBExtract) -> str:
-    return self.binary(expression, "->")
+def arrow_json_extract_sql(self: Generator, expression: exp.JSONExtract) -> str:
+    this = expression.this
+    if self.JSON_TYPE_REQUIRED_FOR_EXTRACTION and isinstance(this, exp.Literal) and this.is_string:
+        this.replace(exp.cast(this.copy(), "json"))
```

Comment:
i think you should make cast default to copy, so you don't need to do this

Reply:
sure